### PR TITLE
Gift Entry:  add DataImport__c.Donation_Donor__c field as a simulated Field Mapping

### PR DESCRIPTION
**Pull request by psadie-sfdo**
# Critical Changes

# Changes
- Simulate DataImport__c.Donation_Donor__c field as a Field Mapping
# Issues Closed

# New Metadata

# Deleted Metadata

### CODE REVIEW

**Comment by psadie-sfdo** [comment]
**lurch: attach W-037936

File: src/classes/BDI_MappingServiceAdvanced.cls [hunk 1]
```diff
@@ -102,21 +102,42 @@ public with sharing class BDI_MappingServiceAdvanced implements BDI_MappingServi
...
+            fieldMappingInstance.setFieldMappingSetName(null);
+            fieldMappingInstance.retrieveCustMetadata(includeUnmappedFields);
         }
         return fieldMappingInstance;
     }
 
     /*******************************************************************************************************
     * @description return singleton instance of the class using specified field mapping set dev name
+    *
+    * @param includeUnmappedFields: Determines whether or not to include simulated field mappings
```
Comment (ptewson-sfdo) [inline src/classes/BDI_MappingServiceAdvanced.cls]: Kind of pedantic, but can we include an @param for fieldMappingSetDevName as well?

File: src/classes/BDI_MappingServiceAdvanced.cls [hunk 2]
```diff
@@ -174,7 +195,7 @@ public with sharing class BDI_MappingServiceAdvanced implements BDI_MappingServi
         }
     }
 
-    private void retrieveCustMetadata(){
+    private void retrieveCustMetadata(Boolean includeUnmappedFields) {
```
Comment (ptewson-sfdo) [inline src/classes/BDI_MappingServiceAdvanced.cls]: is "Unmapped fields" the best name? There are other fields on DI that aren't mapped that are still left out. Maybe utilityFields? Not a critical decision.

File: src/classes/BDI_MappingServiceAdvanced.cls [hunk 3]
```diff
@@ -278,8 +305,87 @@ public with sharing class BDI_MappingServiceAdvanced implements BDI_MappingServi
...
+    * fields for use in the addUnmappedFields method. This is to support adding in fields that aren't
+    * Field Mappings, but need to look like Field Mappings such as the DataImport__c.Donation_Donor__c field.
+    *
+    * @param fieldDescribe: Describe info of the field we need a Field Mapping for.
+    * @param parentObjectMapping: The parent object mapping this Field Mapping needs to reference.
+    */
+    private static BDI_FieldMapping constructSimulatedFieldMapping(Schema.DescribeFieldResult fieldDescribe,
+        BDI_ObjectMapping parentObjectMapping) {
+
+        BDI_FieldMapping fieldMapping   = new BDI_FieldMapping();
```
Comment (ptewson-sfdo) [inline src/classes/BDI_MappingServiceAdvanced.cls]: not like the BIGGEST fan of aligning the equal signs like this but might bet willing to let it slide - not sure if we have guidance around this in our style guide or not

File: src/classes/BDI_MappingServiceAdvanced_TEST.cls [hunk 4]
```diff
@@ -124,12 +124,39 @@ private class BDI_MappingServiceAdvanced_TEST {
         UTIL_CustomSettingsFacade.setDataImportSettings(dis);
 
         //Retrieve a mapping service instance
-        BDI_MappingServiceAdvanced bdiMSAdv = BDI_MappingServiceAdvanced.getInstance(BDI_MappingServiceAdvanced.DEFAULT_DATA_IMPORT_FIELD_MAPPING_SET_NAME);
-        BDI_FieldMappingSet fms = BDI_MappingServiceAdvanced.getFieldMappingSet(BDI_MappingServiceAdvanced.DEFAULT_DATA_IMPORT_FIELD_MAPPING_SET_NAME);
+        BDI_MappingServiceAdvanced bdiMSAdv = BDI_MappingServiceAdvanced.getInstance(BDI_MappingServiceAdvanced.DEFAULT_DATA_IMPORT_FIELD_MAPPING_SET_NAME, true);
```
Comment (ptewson-sfdo) [inline src/classes/BDI_MappingServiceAdvanced_TEST.cls]: Can we avoid these line breaks?

File: src/classes/BDI_MappingServiceAdvanced_TEST.cls [hunk 5]
```diff
@@ -124,12 +124,39 @@ private class BDI_MappingServiceAdvanced_TEST {
...
+
+    /*******************************************************************************************************
+    * @description Tests that the Donation_Donor__c field from DataImport__c object has been added as a
+    * simulated Field Mapping.
+    */
+    @isTest static void shouldAddDonationDonorFieldToMapsAsSimulatedFieldMapping() {
+        //Set the custom setting for Data import to use Data Import Field Mappings (eg Custom Metadata Type Mapping)
+        Data_Import_Settings__c dis = UTIL_CustomSettingsFacade.getDataImportSettings();
+        dis.Field_Mapping_Method__c = BDI_DataImportService.FM_DATA_IMPORT_FIELD_MAPPING;
+        dis.Default_Data_Import_Field_Mapping_Set__c = BDI_MappingServiceAdvanced.DEFAULT_DATA_IMPORT_FIELD_MAPPING_SET_NAME;
```
Comment (ptewson-sfdo) [inline src/classes/BDI_MappingServiceAdvanced_TEST.cls]: Can we avoid these line breaks (also line #148, 149)

File: src/lwc/geTemplateBuilderFormField/geTemplateBuilderFormField.html [hunk 6]
```diff
@@ -46,8 +46,10 @@
                         </template>
 
                         <template if:true={isLightningCombobox}>
-                            <c-ge-form-field-picklist field-name={targetFieldApiName}
+                            <c-ge-form-field-picklist class-name="slds-m-around_none"
```
Comment (ptewson-sfdo) [inline src/lwc/geTemplateBuilderFormField/geTemplateBuilderFormField.html]: Are you starting to include the attributes to support automation that you worked on with Sravani? This comment not actually related to this line)

Comment (psadie-sfdo) [reply src/lwc/geTemplateBuilderFormField/geTemplateBuilderFormField.html]: Not yet in this case. Sravani is whipping up a work item for that and running it by Ali I believe.

**Comment by psadie-sfdo** [comment]
**lurch: attach W-037936

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000001HpVzQAK" target="_blank">W-037936</a>

**Pull request merged**